access/handler.go

```diff
@@ -466,3 +467,128 @@ func blockEventsToMessage(block flow.BlockEvents) (*access.EventsResponse_Result
 		Events:         eventMessages,
 	}, nil
 }
+
+type ExecutionAPIHandler struct {
+	accessAPI API
+	chain     flow.Chain
+}
+
+func NewExecutionAPIHandler(api API, chain flow.Chain) *ExecutionAPIHandler {
+	return &ExecutionAPIHandler{
+		accessAPI: api,
+		chain:     chain,
+	}
+}
+
+func (h *ExecutionAPIHandler) Ping(ctx context.Context, _ *execution.PingRequest) (*execution.PingResponse, error) {
+	err := h.accessAPI.Ping(ctx)
```

Comment:
This should actually return a ` Unsupported Error`. An unstaked AN should not both pinging an Execution node. 
The reason this API exist on the execution API is because we want an ability to ping an access node and check if it is up AND that the execution node and collection node it talks to are also up.

Reply:
This is true, thanks for the catch.